core/domain/html_cleaner.py

```diff
@@ -458,13 +460,127 @@ def convert_to_text_angular(html_data):
+    # there are consecutive ol/ul tags this block adds the margin:left according
+    # to the number of ol/ul tags and then remove these consecutive ol/ul tags
+    # to maintain the CKEditor format.
+    list_tags = ['ol', 'ul']
+    for tag_name in list_tags:
+        for tag in soup.findAll(tag_name):
+            if tag.parent.name in list_tags:
+                prev_sib = tag.previous_sibling
+                if prev_sib and prev_sib.name == 'li':
+                    if 'style' in tag.attrs:
+                        del tag['style']
+                else:
+                    margin_value = 40
+                    if 'style' in tag.attrs:
+                        style = tag['style']
+                        match = re.search('margin-left:[0-9]+px;', style)
+                        margin_value += int(
+                            style[match.start() + 12: match.end() - 3])
+
+                    for child in tag.children:
+                        child_margin_value = margin_value
+                        if 'style' in child.attrs:
+                            style = child['style']
+                            match = re.search('margin-left:[0-9]+px;', style)
+                            child_margin_value += int(
+                                style[match.start() + 12: match.end() - 3])
+                        child['style'] = 'margin-left:%dpx;' % margin_value
+                    tag.unwrap()
+
+    oppia_block_components = [
+        'oppia-noninteractive-image',
+        'oppia-noninteractive-video',
+        'oppia-noninteractive-collapsible',
+        'oppia-noninteractive-tabs'
+    ]
+
+    # Move block components out of p, pre, strong and em tags.
+    for tag_name in oppia_block_components:
+        for tag in soup.findAll(tag_name):
+            while tag.parent.name in ['p', 'pre', 'strong', 'em']:
+                new_parent_for_prev = soup.new_tag(tag.parent.name)
+                new_parent_for_next = soup.new_tag(tag.parent.name)
+                prev_sib = list(tag.previous_siblings)
+                next_sib = list(tag.next_siblings)
+
+                # Previous siblings are accessed in reversed order to
+                # avoid reversing the order of siblings on being wrapped.
+                for sib in reversed(prev_sib):
+                    sib.wrap(new_parent_for_prev)
+
+                for sib in next_sib:
+                    sib.wrap(new_parent_for_next)
+                tag.parent.unwrap()
+
+    # Beautiful soup automatically changes some <br> to <br/>,
+    # so it has to be replaced directly in the string.
+    # Also, when any html string with <br/> is stored in exploration
+    # html strings they are stored as <br>. Since both of these
+    # should match and <br> and <br/> have same working,
+    # so the tag has to be replaced in this way.
+
+    return unicode(soup).replace('<br/>', '<br>')
+
+
+def convert_tag_contents_to_rte_format(html_data, rte_format):
```

Comment:
Instead of using `rte_format`, you can instead accept `rte_conversion_fn` (which will be either `convert_to_text_angular` or `convert_to_ck_editor`) and then just call `rte_conversion_fn` in the body of this function.